src/eth_provider/provider.rs

```diff
@@ -1014,16 +1023,22 @@ where
             }
 
             // Generate primitive transaction, handle error if any
-            let Ok(transaction) = TransactionSignedEcRecovered::try_from(tx.tx.clone()) else {
-                // Delete the pending transaction from the database due conversion error
-                // Malformed transaction
-                self.database
-                    .delete_one::<StoredPendingTransaction>(into_filter("tx.hash", &hash, HASH_HEX_STRING_LEN))
-                    .await?;
-                // Continue to the next iteration of the loop
-                continue;
+            let transaction = match TransactionSignedEcRecovered::try_from(tx.tx.clone()) {
```

Comment:
Why match and not if let else? Doesn't the linter prefer if let else ?

Reply:
Yes if it's a normal `if let else`, the clippy prefers but here I did that to simply catch the error and display it. I told myself that if we had recurring conversion errors it would be a good idea to find out with this method.